src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -709,14 +710,14 @@ public int deflate(ByteBuffer output, int flush) {
             if (input == null) {
                 inputPos = this.inputPos;
                 if (output.isDirect()) {
-                    long outputAddress = ((DirectBuffer) output).address();
+                    var outScope = NIO_ACCESS.acquireSession(output);
                     try {
                         result = deflateBytesBuffer(zsRef.address(),
                             inputArray, inputPos, inputLim - inputPos,
-                            outputAddress + outputPos, outputRem,
+                                ((DirectBuffer) output).address() + outputPos, outputRem,
```

Comment:
Same here, I think I would keep outputAddress to keep the call to deflateByteBuffer simple.